sqlglot/parser.py

```diff
@@ -3781,6 +3778,16 @@ def _parse_if(self) -> t.Optional[exp.Expression]:
 
         return self._parse_window(this)
 
+    def _parse_next_value_for(self) -> t.Optional[exp.Expression]:
+        if not self._match_text_seq("VALUE", "FOR"):
+            return None
```

Comment:
don't you need to retreat because next was consumed?

Reply:
both good points, will address shortly, thanks